src/formatting.rs

```diff
@@ -60,21 +70,47 @@ pub(crate) fn format_inner(contents: &[u8], _config: &Config) -> Vec<u8> {
         trace!("shape contents: {:?}", &content);
 
         match shape {
-            FlatShape::String | FlatShape::Int | FlatShape::Nothing => out.extend(bytes),
+            FlatShape::Int | FlatShape::Nothing => out.extend(bytes),
+            FlatShape::StringInterpolation => {
+                out.extend(bytes);
+            }
             FlatShape::List | FlatShape::Record => {
                 bytes = trim_ascii_whitespace(bytes);
-                let printable = String::from_utf8_lossy(bytes).to_string();
-                trace!("stripped the whitespace, result: {:?}", printable);
                 out.extend(bytes);
             }
+            FlatShape::Block | FlatShape::Closure => {
+                bytes = trim_ascii_whitespace(bytes);
+                out.extend(bytes);
+            }
+            FlatShape::String => {
+                out.extend(bytes);
+                // if it'a string after a `def`, add a space before the `[`
+                if after_a_def {
+                    out.extend(b" ");
+                }
+            }
             FlatShape::Pipe => {
                 out.extend(b"| ");
             }
-            FlatShape::External | FlatShape::ExternalArg => {
+            FlatShape::InternalCall(declid) => {
+                trace!("Called Internal call with {declid}");
+                out = resolve_call(bytes, declid, out);
+                // make the def_name bool true if its a `def` keyword
+                after_a_def = declid == 5;
+            }
+            FlatShape::External => out = resolve_external(bytes, out),
+            FlatShape::ExternalArg | FlatShape::Signature | FlatShape::Keyword => {
+                out.extend(bytes);
+                // out.extend(b" ");
```

Comment:
i think we can get rid of this :wink:

Reply:
Mmmm I don't see why. Do you have a reason? 👍🏼

Reply:
well, that's a piece of dead code, right?
i'm just talking about like 104, not the rest

Reply:
Oh! yeah, that's right! Removing!